addons/external-dns/gcp/credentials/outputs.tf

```diff
@@ -0,0 +1,8 @@
+output "dns_project" {
+  value = local.dns_project
+}
+
+output "dns_credentials" {
```

Comment:
"credentials" function should not be just a credential function, but a subzone create function too with the given outputs, the outputs must be cloud agnostic.

Reply:
as more I think, we should just have a cloud specific `external-dns` function/module that include dependencies management and performs the install, and keeps credentials within the module (with no exposure)

Reply:
I already replied to the subzone part in the previous comment.

For the dependencies part, that's fine. I can move the credentials to `modules/gcp/dns-credentials` if needed.
I wanted to avoid doing a specific per-provider module because it is not needed and it would lead us to duplicate stuff: if we have a `module/gcp/external-dns` (like we have in the `register_fqdn`, we still have to have the per-provide thing in the add-on folder, because module paths cannot be parameterized, so we don't gain anything.

The current approach, instead, already has a completely generic module that requires little inputs and can be reused in every cloud.

Reply:
it is a hard point to argue, as you are implementing it only for GCP stage so it is hard to technically prove it is generic, the easiest way to get to the common, it is to compare and merge the three options when drafted, so if you don't mind - can you please store all the steps under the given `module/gcp/external-dns` folder - technically at least it will account to the general stands of how I am trying to keep this repo.

Reply:
>it is a hard point to argue

then ake your time

>how I am trying to keep this repo.

this repo is no longer about _you_.

Reply:
i love the comments and energy, but @nacx this is not a monorepo, so you cannot lead by authority, so please do take into consideration the fact this is not a toy for you to overtake, but a field facing repo that has a consumers beside you and was built as my personal toy project which appeared to be of use to the people so I will still do :) pet this repo, until I am here so :) consensus and balance that what I need you to account here.